lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -60,6 +60,8 @@ export default function SubmitGradeForm({
     student ? `grade:${student.userid}` : null,
     student ? () => fetchGrade(student) : undefined
   );
+  // Tracks unsaved changes in loaded grade
+  const gradeUnsavedRef = useRef(false);
```

Comment:
Did you consider making this a controlled component? If the component has states for the current value of the grade and the last-saved value of the grade, it would be easy to define an effect that registers a `beforeunload` handler when they differ and removes it when they are the same.